### DISCUSSION TITLE
Dynamic params taken from DB in preview files are not showing until restarting the server

### DISCUSSION THREAD

**Discussion by RaulSeSo** [lookbook-hq/lookbook#374] [Q&A] [answered]
Hi all,

before anything, thanks so much for that amazing tool, it's really good.

I'm using `lookbook 1.3.3` with `ruby 3-0.5` and `rails 6.1`. It is used in both, `development` and `beta` environments.

I'm trying to use [dynamic params](https://lookbook.build/guide/previews/params/) in `_preview.rb` files using database information. To do it I set:

```ruby
#
# @param act_id select {{Act.last(10).pluck(:id)}} "Select one of the latest 10 acts"
#
``` 
Everything works perfectly fine except when new `Act` records are added to the database. When it happens, these new records are not shown in the `dropdown` until I re-start the server (in any of the environments), so I can't select any of the new records before re-starting the server.
Any ideas that could help to deal with that?

Thanks in advance!

**Comment by allmarkedup** [comment, marked as answer]
Hi @RaulSeSo,

So the problem here is that Lookbook only reloads the preview files when it detects a change to the file itself - it doesn't know anything about changes in your database.

It may be possible to trigger a manual update from within an active record callback, although I've never tried that myself. Something like this may work (for v1.3.3):

```rb
class Act < ApplicationRecord
  after_create do
    Lookbook.engine.parser.parse
  end
end
```

Let me know how you get on if you have a chance give it a try 🙂

**Reply by RaulSeSo** [reply]
Hi @allmarkedup!
Thank you so much, it makes sense and it works as you expected (currently in v1.5.3).

In our particular case, adding it to the active record callbacks maybe is a bit excessive because we would like to avoid that kind of dependency for each model where we use Lookbook. On the other hand, it's a piece of code that we don't want to execute in production and I think it would be too elaborate for the case. We could find solutions for these situations, but, you know, maybe it's a bit too much.

However, it's good to know that it was not something that `_preview` files were thought to handle and thank you so much for the workaround, I appreciate that. Do you think that an inbuilt way to handle these situations would be valuable for Lookbook?

**Reply by allmarkedup** [reply]
@RaulSeSo ah glad that it worked but totally understand what you are saying about dependencies. I'm not really sure what other options are though unfortunately 😕

Even if there was an inbuilt solution it would probably have to know about your database and tables and it could get quite complicated quite quickly. If I'm honest I can't see Lookbook ever integrating with a database but maybe thats just because I'm a little allergic to SQL 😝 I'm sorry that doesn't help you very much here though.

If there is anything else I can be of help with just let me know and I'll try to get back to you as soon as possible.

**Reply by RaulSeSo** [reply]
Thanks so much for your time!
